jwcrypto/jwa.py

```diff
@@ -1105,34 +1105,32 @@ class JWA(object):
     }
 
     @classmethod
+    def instantiate_alg(cls, name, use=None):
+        obj = cls.algorithms_registry[name]()
+        if use is not None and obj.algorithm_use != use:
+            raise KeyError
+        return obj
```

Comment:
yeah, I think I had it this way before and then canged it back nd forth a few times, let me change it